app/views/users/show.html.erb

```diff
@@ -24,15 +24,106 @@
     }
     <% end %>
   </style>
-  <div class="user-profile-header" style="<%= user_colors_style(@user) %>" itemscope itemtype="http://schema.org/Person" itemprop="mainEntity">
-    <meta itemprop="url" content="https://dev.to/<%= @user.username %>">
+
+<script type="application/ld+json">
+  {
+    "@context": "http://schema.org",
+    "@type": "Person",
+    "mainEntityOfPage": {
+      "@type": "WebPage",
+      "@id": "https://dev.to/<%= @user.username %>"
+    },
+    "url": "<%= user_url(@user) %>",
+    "image": <%= ProfileImage.new(@user).get(width: 320) %>,
+    "name": "<%= @user.name %>",
+    "description": [
+      {
+        "<%= @user.summary.presence || ["404 bio not found"].sample %>",
+        "sameAs": [
+          <% if @user.mostly_work_with.present? %>
+            "<%= @user.mostly_work_with %>",
+          <% end %>
+          <% if @user.currently_hacking_on.present? %>
+            "<%= @user.currently_hacking_on %>",
+          <% end %>
+          <% if @user.currently_learning.present? %>
+            "<%= @user.currently_learning %>"
+          <% end %>
```

Comment:
this is again tricky as if only `mostly_work_with` for example is present, you end up with a an array lke `["value",]` (see the final comma) which is indeed invalid JSON as the testing tool flags.